### ISSUE TITLE
recurring event completed, no new instances created

### ISSUE THREAD

**Issue by gour** [GothenburgBitFactory/taskwarrior#3]
#### To report a bug...

* What command(s) did you run?
I did complete a recurring  task (eb5dbdbd)
* What did you expect to happen?

To complete and new instance created.

* What actually happened?

The task's status changed to `completed` and no new instances were created.

* Paste the output of the ```task diag``` command.
```
$ task diag

task 2.5.1
   Platform: Linux

Compiler
    Version: 7.1.1 20170718 (Red Hat 7.1.1-6)
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      Built: Aug  5 2017 00:01:30
      CMake: 3.9.0
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.5.14
 Build type: 

Configuration
       File: /home/gour/.taskrc (found), 2535 bytes, mode 100644
       Data: /home/gour/task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
  rc.editor: /usr/bin/vim
     Server: atmarama.dtdns.net:6544
         CA: /home//gour/task/ca.cert.pem, readable, 3720 bytes
      Trust: strict
Certificate: /home/gour/task/gour.cert.pem, readable, 3710 bytes
        Key: /home/gour/task/gour.key.pem, readable, 24745 bytes
    Ciphers: NORMAL
      Creds: Atmarama/gour/************************************

Hooks
     System: Enabled
   Location: /home/gour/task/hooks
     Active: on-add-pirate    (executable)
             on-modify-pirate (executable)
   Inactive: 

Tests
      $TERM: xterm-256color (324x64)
       Dups: Scanned 5388 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 5388 tasks for broken references:
             No broken references found
```

Here is the 'info' output in regard to the task  involved:

```
tasksh> eb5dbdbd info
[task eb5dbdbd info]

Name          Value                               
ID            -
Description   zalijevanje Arjune                  
Status        Completed
Project       home.duties                         
Recurrence    3days
Parent task   4d534d7e-bf0c-4dbc-94bc-89258ae8b157
Mask Index    18.000000
Entered       2018-01-29 10:54:09 (10d)           
Due           2018-02-01 10:00:00
End           2018-01-31 19:42:45                 
Last modified 2018-01-31 19:42:46 (7d)
Virtual tags  CHILD COMPLETED UNBLOCKED PROJECT   
UUID          eb5dbdbd-864d-475b-bf8e-193760a3c89d
Urgency       13.05                               

    project      1 *    1 =      1
    due          1 *   12 =     12
    age      0.027 *    2 =  0.055
                            ------
                             13.05

Date                Modification                                 
2018-01-31 19:42:46 End set to '2018-01-31 19:42:45'.
                    Status changed from 'pending' to 'completed'.
```

and here is the 'info' about the parent task (4d534d7e):

```
tasksh> 4d534d7e info
[task 4d534d7e info]

Name          Value                               
ID            107
Description   zalijevanje Arjune                  
Status        Recurring
Project       home.duties                         
Recurrence    3days
Mask          +++++++++++++++++++++-              
Entered       2017-12-09 08:52:58 (8w)
Due           2017-12-09 10:00:00                 
Last modified 2018-02-07 16:14:51 (21h)
Virtual tags  PARENT UNBLOCKED PROJECT            
UUID          4d534d7e-bf0c-4dbc-94bc-89258ae8b157
Urgency       13.33                               

    project      1 *    1 =      1
    due          1 *   12 =     12
    age      0.167 *    2 =  0.334
                            ------
                             13.33

Date                Modification                                                           
2017-12-09 08:53:23 Project set to 'home.duties'.
                    Recur set to '3days'.
                    Status changed from 'pending' to 'recurring'.
2017-12-09 08:53:28 Mask set to '-'.                                                       
2017-12-09 10:52:20 Mask changed from '-' to '--'.
2017-12-09 15:37:45 Mask changed from '--' to '+-'.                                        
2017-12-12 10:21:56 Mask changed from '+-' to '+--'.
2017-12-12 12:33:36 Mask changed from '+--' to '++-'.                                      
2017-12-15 14:03:18 Mask changed from '++-' to '++--'.
2017-12-16 12:29:20 Mask changed from '++--' to '+++-'.                                    
2017-12-18 10:08:37 Mask changed from '+++-' to '+++--'.
2017-12-18 13:31:13 Mask changed from '+++--' to '++++-'.                                  
2017-12-21 10:01:09 Mask changed from '++++-' to '++++--'.
2017-12-21 11:34:49 Mask changed from '++++--' to '+++++-'.                                
2017-12-24 10:36:44 Mask changed from '+++++-' to '++++++'.
2017-12-24 10:36:46 Mask changed from '++++++' to '++++++-'.                               
2017-12-27 10:37:37 Mask changed from '++++++-' to '++++++--'.
2017-12-28 21:07:50 Mask changed from '++++++--' to '+++++++-'.                            
2017-12-30 09:10:15 Mask changed from '+++++++-' to '++++++++'.
2017-12-30 10:44:21 Mask changed from '++++++++' to '++++++++-'.                           
2018-01-02 10:05:24 Mask changed from '++++++++-' to '++++++++--'.
2018-01-02 12:41:32 Mask changed from '++++++++--' to '+++++++++-'.                        
2018-01-05 10:07:28 Mask changed from '+++++++++-' to '+++++++++--'.
2018-01-06 16:19:11 Mask changed from '+++++++++--' to '++++++++++-'.                      
2018-01-08 10:16:13 Mask changed from '++++++++++-' to '++++++++++--'.
2018-01-08 10:40:36 Mask changed from '++++++++++--' to '+++++++++++-'.                    
2018-01-11 10:15:02 Mask changed from '+++++++++++-' to '+++++++++++--'.
2018-01-12 22:24:51 Mask changed from '+++++++++++--' to '++++++++++++-'.                  
2018-01-14 10:30:10 Mask changed from '++++++++++++-' to '++++++++++++--'.
2018-01-14 20:19:12 Mask changed from '++++++++++++--' to '+++++++++++++-'.                
2018-01-17 10:44:44 Mask changed from '+++++++++++++-' to '+++++++++++++--'.
2018-01-17 13:46:20 Mask changed from '+++++++++++++--' to '++++++++++++++-'.              
2018-01-20 15:57:02 Mask changed from '++++++++++++++-' to '++++++++++++++--'.
2018-01-20 21:16:27 Mask changed from '++++++++++++++--' to '+++++++++++++++-'.            
2018-01-23 10:10:28 Mask changed from '+++++++++++++++-' to '+++++++++++++++--'.
2018-01-23 10:13:56 Mask changed from '+++++++++++++++--' to '++++++++++++++++-'.          
2018-01-26 10:09:18 Mask changed from '++++++++++++++++-' to '++++++++++++++++--'.
2018-01-26 10:11:15 Mask changed from '++++++++++++++++--' to '+++++++++++++++++-'.        
2018-01-29 10:54:10 Mask changed from '+++++++++++++++++-' to '+++++++++++++++++--'.
2018-01-29 12:01:57 Mask changed from '+++++++++++++++++--' to '++++++++++++++++++-'.      
2018-01-31 19:42:46 Mask changed from '++++++++++++++++++-' to '+++++++++++++++++++'.
2018-02-01 10:43:00 Mask changed from '+++++++++++++++++++' to '+++++++++++++++++++-'.     
2018-02-04 10:28:54 Mask changed from '+++++++++++++++++++-' to '+++++++++++++++++++--'.
2018-02-04 10:39:04 Mask changed from '+++++++++++++++++++--' to '++++++++++++++++++++-'.  
2018-02-07 10:24:55 Mask changed from '++++++++++++++++++++-' to '++++++++++++++++++++--'.
2018-02-07 16:14:51 Mask changed from '++++++++++++++++++++--' to '+++++++++++++++++++++-'.
```

so I wonder if you can detect something suspicious?

**Comment by dathanb** [comment]
I'm looking at a first issue to pick up, and considering this one, so I want to understand what's going on. 

Looks like a recurrence template with a recurrence window of 3 days was created. 
A task was created on 2018-01-29 10:54:10 with a due date of 2018-02-01 10:00:00. 
It was closed on 2018-01-31 19:42:46.
At 2018-02-01 10:43:00 a new task was created.

That all seems like it worked correctly? Am I missing something?
